# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/tests/gtest/MockWidget.h
# Commit: dd00e2da3a0f
# Full Hash: dd00e2da3a0f11a7cc0af9e5ece21a339f4393eb
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 69cc1de6f754 -r dd00e2da3a0f gfx/tests/gtest/MockWidget.h
--- a/gfx/tests/gtest/MockWidget.h	Wed Jun 10 21:57:19 2020 +0000
+++ b/gfx/tests/gtest/MockWidget.h	Wed Jun 10 22:21:02 2020 +0000
@@ -33,13 +33,10 @@
 
   void* GetNativeData(uint32_t aDataType) override {
     if (aDataType == NS_NATIVE_OPENGL_CONTEXT) {
-      mozilla::gl::SurfaceCaps caps = mozilla::gl::SurfaceCaps::ForRGB();
-      caps.preserve = false;
-      caps.bpp16 = false;
       nsCString discardFailureId;
       RefPtr<GLContext> context = GLContextProvider::CreateOffscreen(
-          IntSize(mCompWidth, mCompHeight), caps,
-          CreateContextFlags::REQUIRE_COMPAT_PROFILE, &discardFailureId);
+          {mCompWidth, mCompHeight},
+          {CreateContextFlags::REQUIRE_COMPAT_PROFILE}, &discardFailureId);
       return context.forget().take();
     }
     return nullptr;